src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp

```diff
@@ -0,0 +1,442 @@
+    params.setMinFloat("peak_height_min", 0.0);
+    params.setValue("peak_height_max", 4e6, "Used in pickSpectrum(), a peak's intensity needs to be <= peak_height_max_ for it to be picked.");
+    params.setMinFloat("peak_height_max", 0.0);
+    params.setValue("fwhm_threshold", 0.0, "Used in pickSpectrum(), a peak's FWHM needs to be >= fwhm_threshold_ for it to be picked.");
+    params.setMinFloat("fwhm_threshold", 0.0);
+
+    params.setValue("tic_weight", 1.0, "TIC weight when scoring spectra.");
+    params.setMinFloat("tic_weight", 0.0);
+    params.setValue("fwhm_weight", 1.0, "FWHM weight when scoring spectra.");
+    params.setMinFloat("fwhm_weight", 0.0);
+    params.setValue("snr_weight", 1.0, "SNR weight when scoring spectra.");
+    params.setMinFloat("snr_weight", 0.0);
+  }
+
+  void TargetedSpectraExtractor::annotateSpectra(
+    const std::vector<MSSpectrum>& spectra,
+    const TargetedExperiment& targeted_exp,
+    std::vector<MSSpectrum>& annotated_spectra,
+    FeatureMap& features
+  )
+  {
+    annotated_spectra.clear();
+    features.clear(true);
+    const std::vector<ReactionMonitoringTransition> transitions = targeted_exp.getTransitions();
+    for (UInt i=0; i<spectra.size(); ++i)
+    {
+      MSSpectrum spectrum = spectra[i];
+      const double divisor = (String)param_.getValue("rt_unit") == "seconds" ? 1.0 : 60.0;
+      const double spectrum_rt = spectrum.getRT() / divisor;
+      const double rt_left_lim = spectrum_rt - rt_window_ / divisor / 2.0;
+      const double rt_right_lim = spectrum_rt + rt_window_ / divisor / 2.0;
+      const std::vector<Precursor> precursors = spectrum.getPrecursors();
+      if (precursors.size() < 1)
+      {
+        throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION,
+                                         "Spectrum does not contain precursor info.");
+      }
+      const double spectrum_mz = precursors[0].getMZ();
+      const double mz_tolerance = mz_unit_is_Da_ ? mz_tolerance_ : mz_tolerance_ / 1e6;
+      const double mz_left_lim = spectrum_mz - mz_tolerance;
+      const double mz_right_lim = spectrum_mz + mz_tolerance;
+
+      LOG_DEBUG << "[" << i << "]\trt: " << spectrum_rt << "\tmz: " << spectrum_mz << std::endl;
+
+      for (UInt j=0; j<transitions.size(); ++j)
+      {
+        const TargetedExperimentHelper::Peptide peptide = targeted_exp.getPeptideByRef(transitions[j].getPeptideRef());
+        double target_rt;
+        if (!peptide.rts.empty() &&
+            peptide.rts[0].hasCVTerm("MS:1000895") &&
+            !peptide.rts[0].getCVTerms()["MS:1000895"].empty())
+        {
+          target_rt = peptide.rts[0].getCVTerms()["MS:1000895"][0].getValue().toString().toDouble();
+        }
+        else
+        {
+          target_rt = peptide.getRetentionTime(); // accesses to MS:1000896. TODO: update if "guesstimate" in TargetedExperimentHelper.h is fixed
+        }
```

Comment:
this is fixed in #3043

Reply:
I did not think of that, thanks! I also prefer to keep everything in seconds since OpenMS does that by default.